events/utils.py

```diff
@@ -38,3 +44,38 @@ def update(d, u):
         else:
             d[k] = u[k]
     return d
+
+
+def recache_n_events(keywords):
+    """
+    Recache the number of events for the given keywords.
+
+    :type keywords: Iterable[Keyword]
+    """
+
+    # The helper function has to exist outside the model, because it is used in migration.
+    # Django apps cannot serialize unbound instance functions when saving model history during migration.
+    if not keywords:
+        return
+    with transaction.atomic():
+        keywords.update(n_events=0)
+        for keyword_id, n_events in count_events_for_keywords((k.id for k in keywords)).items():
+            keywords.filter(id=keyword_id).exclude(n_events=n_events).update(n_events=n_events,
```

Comment:
Also, it's not a good idea to use `keywords.filter(id=keyword_id)`, since `keywords` could be a queryset that has 1000 other filter clauses already (and you'll end up making an SQL query that looks like `UPDATE ... WHERE id=... AND 1000-other-clauses`.

Just use `Keyword.objects.filter()`.